rust/reflection/src/lib.rs

```diff
@@ -54,6 +56,12 @@ pub enum FlatbufferError {
     TypeNotSupported(String),
     #[error("No type or invalid type found in union enum")]
     InvalidUnionEnum(),
+    #[error("RwLock read/write error: {0}")]
+    RwLockPoisonError(String),
+    #[error("Table or Struct doesn't belong to the buffer")]
+    InvalidTableOrStruct(),
```

Comment:
Nit: if we don't need any data inside, we can skip the `()` and make it just `InvalidTableOrStruct,` here.